starter_code/src/movies.js

```diff
@@ -1,19 +1,108 @@
+            return accumulator + parseFloat(movie.rate);
+        }, 0) / moviesArr.length;
+    avgRate = parseFloat(avgRate.toFixed(2));
+    return avgRate;
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
-
-
+function dramaMoviesRate(moviesArr) {
+    const dramaSelection = moviesArr.filter(movie =>
+        movie.genre.includes("Drama")
+    );
+    if (!dramaSelection.length) return 0;
+    return ratesAverage(dramaSelection);
+}
 // Iteration 3: Ordering by duration - Order by time duration, ascending (in growing order)
+function orderByDuration(moviesArr) {
+    const moviesCopy = moviesArr.slice();
+    if (moviesArr.length === 1) return moviesArr;
 
+    //
+    for (movie of moviesCopy) {
+        if (typeof movie.duration === "string") {
+            const time = movie.duration.split(" ");
+            const hours = parseInt(time[0]);
+            const minutes = parseInt(time[1]) || 0;
+            const duration = hours * 60 + minutes;
+            movie.duration = duration;
+        }
+    }
 
-// Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
+    const sortedByDuration = moviesCopy.sort((a, b) => {
+        if (a.duration === b.duration) {
+            return a.title.localeCompare(b.title);
+        }
+        return a.duration - b.duration;
+    });
+    return sortedByDuration;
+}
 
+// Iteration 4: Steven Spielberg. The best? - How many movies did STEVEN SPIELBERG direct
+function howManyMovies(moviesArr) {
+    const stevensDramas = moviesArr.filter(movie => {
+        return (
+            movie.genre.includes("Drama") && movie.director === "Steven Spielberg"
+        );
+    });
+    return stevensDramas.length;
+}
 
 // Iteration 5: Alphabetic Order - Order by title and print the first 20 titles
+function orderAlphabetically(moviesArr) {
+    let alphaOrder = moviesArr.sort((a, b) => {
+        return a.title.localeCompare(b.title);
+    });
+    alphaOrder = alphaOrder.map(movie => movie.title).slice(0, 20);
+    return alphaOrder;
+}
 
 // Iteration 6: Time Format - Turn duration of the movies from hours to minutes
+function turnHoursToMinutes(moviesArr) {
+    let moviesDurationCorrected = moviesArr.map(movie => {
+        let time = movie.duration.split(" ");
+        let duration = 0;
+
+        if (time.length >= 2) {
+            const hours = parseInt(time[0]) || 0;
+            const minutes = parseInt(time[1]) || 0;
+            duration = hours * 60 + minutes;
+        } else if (time[0].indexOf("min") !== -1) {
+            duration = parseInt(time[0]) || 0; //duration = minutes
+        } else {
+            duration = parseInt(time[0]) * 60 || 0; // duration = hours*60
+        }
+        return {
+            ...movie,
+            duration
+        };
+    });
 
+    return moviesDurationCorrected;
+}
 // BONUS Iteration: Best yearly rate average - Best yearly rate average
+function bestYearAvg(moviesArr) {
+    let bestYear = 1900,
```

Comment:
hacky! Create an array with all the distinct years instead.
Other than that, good job!